src/Components/Components/src/Routing/RouteTableFactory.cs

```diff
@@ -114,6 +114,11 @@ private static string[] GetParameterNames(RouteTemplate routeTemplate)
         /// </summary>
         internal static int RouteComparison(RouteEntry x, RouteEntry y)
         {
+            if (ReferenceEquals(x, y))
+            {
+                return 0;
+            }
+
```

Comment:
We should follow up here with a better fix for this overall. Throwing from the comparer isn't really a good design for this, this is kinda a workaround.